### DISCUSSION TITLE
Option to turn off auto refresh, or otherwise temporarily suspend it

### DISCUSSION THREAD

**Discussion by MagixDragon** [derceg/explorerplusplus#469] [Ideas]
If I am working in a directory trying to rename files while a download is going on in that directory (so other files are being added, deleted, or modified in the folder), when the auto refresh kicks in every second or so, I lose the in progress edit of the file name.  I can work around it by pulling up the properties dialog and doing it that way.  Also, because the file list tries to stay sorted, files jump around.

In Windows Explorer the problem isn't as severe, I can hit F2 to edit a file name and that edit won't be cancelled on a notification of a folder update.  Part of that is probably because it adds new files to the end and isn't recalculating and refreshing the column width.

I do think being able to toggle off the folder refresh has the potential to confuse some users, but maybe there is a way to disable the folder update in the UI while an item is being edited?  Or at least the part that causes the edit to be canceled.

If that's not clear, and if I get some time, I'll come back and add a short video if possible comparing Explorer++ and Windows Explorer behavior.

**Comment by AgentSam** [comment]
> If that's not clear...

## GOOD BASE IDEA

The base idea is quite clear. That said, I'm sure it would benefit everyone to have a bit more in-depth discussion of what's the best way to go about implementing this.

## SHORT TERM -- IMPLEMENT AS A TOGGLE?

I think, that initially, I would perfer to see this implemented as a keyboard shortcut toggle:
- Toggle "Auto Refresh" OFF ... while doing renaming and whatever.
- Toggle "Auto Refresh" ON .... back to having file&folder pane auto-refresh enabled.

## LONG TERM FUTURE IDEAS -- EVENTS & ACTIONS

It might be interesting, if EPP had a way to allow user-customizable "events" for various operations, and each "event" could have associated "actions". This way we could implement "workflows" that best suit our particular situations.

To use your "file-pane auto refresh" as a case-example:
- One event could be a keypress such as "ctrl+u" (that's just an example,), to trigger an action like "set file auto-refresh off"..
- Another event could be triggered by "new folder selected", which might run an action like "set file auto-refresh on".

The result of these events would be, that you can toggle filelist updates off whenever you like (via the keyboard shortcut), but the auto-refresh would be re-enabled simply by changing folders. And if that's not desirable, easy to change by adjusting the event.

Sidenote: Events and actions would require their own macro-language, or be implemented using some embedded scripting language. But I think Derceg has already been experimenting with such! I think this would bring ultimate workflow automation and customization.

**Reply by MagixDragon** [reply]
One thing I might not have tried when struggling with this was the "Insert new items in their sorted positions" checkbox in the "Files and Folders" Options, if it was there in the version I was using I may not have tried unchecking it.  Another solution, if the listview object being used supports it, would be to not allow the shell change notification handler to take focus away from a ListItem that is being edited. 

I would hope any feature isn't implemented solely as a "keyboard shortcut" but would have corresponding menu and toolbar access.
